src/zkevm_specs/evm/execution/jump.py

```diff
@@ -0,0 +1,27 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+
+
+def jump(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.JUMP)
+
+    # Do not check 'dest' is within MaxCodeSize(24576) range in successful case
+    # as byte code lookup can ensure it.
+    dest = instruction.stack_pop()
+
+    # Get `dest` raw value in max three bytes
+    bytes = instruction.rlc_to_bytes(dest, 32)
+    dest_value = instruction.bytes_to_int(bytes[:3])
```

Comment:
When we want to strictly limit random linear combination to `n` word, we use exact amount of bytes to recomposite it.
```suggestion
    dest_value = instruction.bytes_to_int(instruction.rlc_to_bytes(dest, 8))
```